src/main/java/it/gov/pagopa/payhub/auth/mypay/service/MyPayOperatorsService.java

```diff
@@ -0,0 +1,38 @@
+package it.gov.pagopa.payhub.auth.mypay.service;
+
+import it.gov.pagopa.payhub.auth.mypay.model.MyPayOperator;
+import it.gov.pagopa.payhub.auth.mypay.repository.MyPayOperatorsRepository;
+import java.util.Optional;
+import java.util.Set;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MyPayOperatorsService {
+
+  private final MyPayOperatorsRepository myPayOperatorsRepository;
+
+  public MyPayOperatorsService(MyPayOperatorsRepository myPayOperatorsRepository){
+    this.myPayOperatorsRepository = myPayOperatorsRepository;
+  }
+
+  public void registerMyPayOperator(String mappedExternalUserId,String email, String organizationIpaCode,
+      Set<String> roles) {
+
+    Optional<MyPayOperator> existingMyPayOperator = Optional.ofNullable(
+        myPayOperatorsRepository.findByCodFedUserIdAndCodIpaEnte(mappedExternalUserId, organizationIpaCode));
+    //if exist update else insert
+    if(existingMyPayOperator.isPresent()) {
+      MyPayOperator operator = existingMyPayOperator.get();
+      operator.setRuolo(roles.stream().findFirst().orElse(null));
```

Comment:
I don't agree on "findFirst". In MyPay the "ruolo" field is filled:
* with ROLE_ADMIN if the role list contains it
* null if the list contains ROLE_OPERATOR
* the user is not stored if the list doesn't contain neither of the 2 past